to.etc.domui/src/main/java/to/etc/domui/component/wizard/WizardPopupWindow.java

```diff
@@ -0,0 +1,193 @@
+
+	private static final String DEFAULT_CANCEL_BUTTON_LABEL = BUNDLE.getString("wizardstep.default.cancelbutton");
+
+	private static final String DEFAULT_END_BUTTON_LABEL = BUNDLE.getString("wizardstep.default.endbutton");
+
+	private static final String DEFAULT_BACK_BUTTON_LABEL = BUNDLE.getString("wizardstep.default.backbutton");
+
+	private static final String STEP_LABEL_NAME = "name";
+
+	private static final String STEP_LABEL_CANCEL = "cancel";
+
+	private static final String STEP_LABEL_BACK = "back";
+
+	private static final String STEP_LABEL_NEXT = "next";
+
+	private static final String STEP_LABEL_END = "end";
+
+	private final ArrayList<WizardPopupStep> m_steps = new ArrayList<>();
+
+	private WizardPopupStep m_currentStep;
+
+	private Div m_currentStepBody;
+
+	public WizardPopupStep getCurrentStep() {
+		return m_currentStep;
+	}
+
+	public void setCurrentStep(@Nonnull WizardPopupStep step) {
+		m_currentStep = step;
+	}
+
+	public ArrayList<WizardPopupStep> getSteps() {
+		return m_steps;
+	}
+
+	/**
+	 * WizardPopupWindow's (i.e. 'wizard's') constructor. Setup the wizard is easy. First, provide a valid String title. Next,
+	 * provide the steps that the wizard has to display by adding a Map with step labels (to create those labels, one can use
+	 * the static method createWizardStepLabels in this class), and by adding an implementation of {@link IWizardPopupStep}.
+	 * @param wizardTitle
+	 * @param steps
+	 */
+	public WizardPopupWindow(@Nonnull String wizardTitle, @Nonnull Map<Map<String, String>, IWizardPopupStep> steps) {
+		super(true, true, 1280, 800, wizardTitle);
+		setupWizard(steps);
+	}
+
+	/**
+	 * This setups the wizard with the values provided in the constructor.
+	 * @param steps
+	 */
+	private void setupWizard(@Nonnull Map<Map<String, String>, IWizardPopupStep> steps) {
+		if(steps.size() <= 0) {
+			throw new IllegalStateException("At least one phase must be defined!");
+		}
+		for(Map<String, String> step : steps.keySet()) {
+			m_steps.add(new WizardPopupStep(step.get(STEP_LABEL_NAME), steps.get(step), step.get(STEP_LABEL_BACK), step.get(STEP_LABEL_NEXT), step.get(STEP_LABEL_CANCEL), step.get(STEP_LABEL_END)));
+		}
+		m_currentStep = m_steps.get(0);
+	}
+
+	@Override
+	public void createContent() throws Exception {
+		refreshNavbarContainer();
+		refreshBodyContainer();
+	}
+
+	/**
+	 * This refreshes the div that holds the wizard's navbar.
+	 */
+	public void refreshNavbarContainer() {
+		Div navbar = buildNavbar();
+		Div stepsContainer = buildStepsContainer();
+		navbar.add(stepsContainer);
+		add(navbar);
+	}
+
+	/**
+	 * This refreshes the div that holds the body of the current step.
+	 * @throws Exception
+	 */
+	public void refreshBodyContainer() throws Exception {
+		if(null == m_currentStepBody) {
+			m_currentStepBody = new Div();
+		}
+		m_currentStepBody.removeAllChildren();
+		m_currentStepBody.add(new WizardPopupStepBodyFragment(m_currentStep, getActions(), isFirstStep(), isLastStep()));
```

Comment:
You should not have to new anything here; the steps should already be there somehow? What happens when you go forth and back?